addon/components/au-date-picker.js

```diff
@@ -1,93 +1,142 @@
+import { assert, runInDebug } from '@ember/debug';
+import { guidFor } from '@ember/object/internals';
+
+const DEFAULT_ADAPTER = {
+  parse: function(value = '', createDate) {
+    const matches = value.match(/^(\d{1,2})-(\d{1,2})-(\d{4})$/);
+    if (matches) {
+      return createDate(matches[3], matches[2], matches[1]);
+    }
+  },
+  format: function(date) {
+    return `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`;
+  },
+};
+
+const DEFAULT_LOCALIZATION = {
+  buttonLabel: 'Choose date',
+  placeholder: 'DD-MM-YYYY',
+  selectedDateMessage: 'Selected date is',
+  prevMonthLabel: 'Previous month',
+  nextMonthLabel: 'Next month',
+  monthSelectLabel: 'Month',
+  yearSelectLabel: 'Year',
+  closeLabel: 'Close window',
+  keyboardInstruction: 'You can use arrow keys to navigate dates',
+  calendarHeading: 'Choose a date',
+  dayNames: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+  monthNames: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
+  monthNamesShort: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
+};
 
 export default class AuDatePickerComponent extends Component {
+  @isoDate value;
+  @isoDate min;
+  @isoDate max;
 
-  //Default Localization
-  @tracked localization = {
-    buttonLabel: 'Choose date',
-    placeholder: 'DD-MM-YYYY',
-    selectedDateMessage: 'Selected date is',
-    prevMonthLabel: 'Previous month',
-    nextMonthLabel: 'Next month',
-    monthSelectLabel: 'Month',
-    yearSelectLabel: 'Year',
-    closeLabel: 'Close window',
-    keyboardInstruction: 'You can use arrow keys to navigate dates',
-    calendarHeading: 'Choose a date',
-    dayNames: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
-    monthNames: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
-    monthNamesShort: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
-  };
+  get adapter() {
+    if (!this.args.adapter) {
+      return DEFAULT_ADAPTER;
+    }
 
-  // Default Adapter
-  @tracked adapter = {
-    parse: function(value = '', createDate) {
-      const matches = value.match(/^(\d{1,2})\.(\d{1,2})\.(\d{4})$/);
-      if (matches) {
-        return createDate(matches[3], matches[2], matches[1]);
-      }
-    },
-    format: function(date) {
-      return `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`;
-    },
-  };
+    runInDebug(() => validateAdapter(this.args.adapter));
 
-  constructor(owner, args) {
-    super(owner, args);
-    this._assignLocalization(args);
-    this._assignAdapter(args);
+    return {
+      ...DEFAULT_ADAPTER,
+      ...this.args.adapter,
+    };
   }
 
-  @action _assignLocalization(args){
+  get id() {
+    return this.args.id ? this.args.id : guidFor(this);
```

Comment:
I guess this would also benefit from the unique-id polyfill that you've introduced in GN?

Reply:
Yea, but then this addon would need to ship that polyfill as well (and do the conditional logic in the template), so I decided against it since it's simple enough. It might be worth it if other components do something similar, but I would have to check.